packages/utilities/src/CliUtils.ts

```diff
@@ -126,7 +342,204 @@ export class CliUtils {
         }
         const numDashes = header.length + 1;
         const headerText = TextUtils.formatMessage("{{indent}}{{headerText}}\n{{indent}}{{dashes}}",
-            { headerText: header.toUpperCase(), dashes: Array(numDashes).join("-"), indent });
+            {headerText: header.toUpperCase(), dashes: Array(numDashes).join("-"), indent});
         return TextUtils.chalk[color](headerText);
     }
+
+
+    /**
+     * Accepts an option name, and array of option aliases, and their value
+     * and returns the arguments style object.
+     *
+     * @param {string} optName - The command option name, usually in kebab case (or a single word)
+     *
+     * @param {string[]} optAliases - An array of alias names for this option
+     *
+     * @param {*} value - The value to assign to the argument
+     *
+     * @returns {ICommandArguments["args"]} - The argument style object
+     *
+     * @example <caption>Create Argument Object</caption>
+     *
+     * CliUtils.setOptionValue("my-option", ["mo", "o"], "value");
+     *
+     * // returns
+     * {
+     *    "myOption": "value",
+     *    "my-option": "value",
+     *    "mo": "value",
+     *    "o": "value"
+     * }
+     *
+     */
+    public static setOptionValue(optName: string, optAliases: string[], value: any): ICommandArguments["args"] {
+        let names: IOptionFormat = CliUtils.getOptionFormat(optName);
+        const args: ICommandArguments["args"] = {};
+        args[names.camelCase] = value;
+        args[names.kebabCase] = value;
+        for (const optAlias of optAliases) {
+            if (optAlias.length === 1) {
+                // for single character aliases, set the value using the alias verbatim
+                args[optAlias] = value;
+            } else {
+                names = CliUtils.getOptionFormat(optAlias);
+                args[names.camelCase] = value;
+                args[names.kebabCase] = value;
+            }
+        }
+        return args;
+    }
+
+
+    /**
+     * Accepts the yargs argument object and constructs the base imperative
+     * argument object. The objects are identical to maintain compatibility with
+     * existing CLIs and plugins, but the intent is to eventually phase out
+     * having CLIs import anything from Yargs (types, etc).
+     *
+     * @param {Arguments} args - Yargs argument object
+     *
+     * @returns {ICommandArguments} - Imperative argument object
+     *
+     */
+    public static buildBaseArgs(args: Arguments): ICommandArguments {
+        const impArgs: ICommandArguments = {...args};
+        Object.keys(impArgs).forEach((key) => {
+            if (key !== "_" && key !== "$0" && impArgs[key] === undefined) {
```

Comment:
Would you want to remove `null` and `''` (empty strings) as well?
Maybe use `!impArgs[key]` if that's the case :smile:
What about nested null properties? (would you want to recursively remove them too?)
Here is a [stackoverflow comment](https://stackoverflow.com/a/38340374/9937473) than might be interesting :yum: